src/tests/async/reflection/reflection-simple.cs

```diff
@@ -19,15 +21,161 @@ public static void TestEntryPoint()
+            $"{map.InterfaceMethods[0]?.ToString()} --> {map.TargetMethods[0]?.ToString()}");
+
+        Assert.Equal("System.Threading.Tasks.Task TReturning() --> System.Threading.Tasks.Task TReturning()",
+            $"{map.InterfaceMethods[1]?.ToString()} --> {map.TargetMethods[1]?.ToString()}");
+
+        Type structType = typeof(ExampleStruct);
+
+        map = structType.GetInterfaceMap(interfaceType);
+        Assert.Equal(2, map.InterfaceMethods.Length);
+        Assert.Equal("System.Threading.Tasks.Task TaskReturning() --> System.Threading.Tasks.Task TaskReturning()",
+            $"{map.InterfaceMethods[0]?.ToString()} --> {map.TargetMethods[0]?.ToString()}");
+
+        Assert.Equal("System.Threading.Tasks.Task TReturning() --> System.Threading.Tasks.Task TReturning()",
+            $"{map.InterfaceMethods[1]?.ToString()} --> {map.TargetMethods[1]?.ToString()}");
+    }
+
+    [Fact]
+    public static void DynamicAsyncMethod()
+    {
+        //  we will be compiling a dynamic vesion of this method
+        //
+        //  public async static Task StaticMethod(Task arg)
+        //  {
+        //    await arg;
+        //  }
+
+        // Define a dynamic assembly and module
+        AssemblyName assemblyName = new AssemblyName("DynamicAssembly");
+        AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.Run);
+        ModuleBuilder moduleBuilder = assemblyBuilder.DefineDynamicModule("DynamicModule");
+
+        // Define a type
+        TypeBuilder typeBuilder = moduleBuilder.DefineType("DynamicType", TypeAttributes.Public);
+
+        // Define a method
+        MethodBuilder methodBuilder = typeBuilder.DefineMethod(
+            "DynamicMethod",
+            MethodAttributes.Public | MethodAttributes.Static,
+            typeof(Task),
+            new Type[] { typeof(Task) });
+
+        // Set `MethodImpl.Async` flag
+        methodBuilder.SetImplementationFlags(MethodImplAttributes.Async);
+
+        // {
+        //   Await(arg_0);
+        //   ret;
+        // }
+        ILGenerator ilGenerator = methodBuilder.GetILGenerator();
+        ilGenerator.Emit(OpCodes.Ldarg_0);
+#pragma warning disable SYSLIB5007 // 'System.Runtime.CompilerServices.AsyncHelpers' is for evaluation purposes only
+        var mi = typeof(System.Runtime.CompilerServices.AsyncHelpers).GetMethod("Await", BindingFlags.Static | BindingFlags.Public, new Type[] { typeof(Task) })!;
+#pragma warning restore SYSLIB5007
+        ilGenerator.EmitCall(OpCodes.Call, mi, new Type[] { typeof(Task) });
+        ilGenerator.Emit(OpCodes.Ret);
+
+        // Create the type and invoke the method
+        Type dynamicType = typeBuilder.CreateType();
+        MethodInfo dynamicMethod = dynamicType.GetMethod("DynamicMethod");
+        var del = dynamicMethod.CreateDelegate<Func<Task, Task>>();
+
+        // the following should not crash
+        del(Task.CompletedTask);
+        del(FooTask());
+    }
```

Comment:
One more interesting scenario are UnsafeAccessor methods (https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafeaccessorattribute).

We may want to add a test to validate that UnsafeAccessors work fine for Task-returning methods. I think it should just work, but it would not hurt to have a test case for it.

Reply:
Added a test for UnsafeAccessor. Seems to be working as-is